// react-device-mockup/dist/android-mockup/AndroidMockup.js
import { __assign } from "tslib";
import { jsx as _jsx, jsxs as _jsxs, Fragment as _Fragment } from "react/jsx-runtime";
import { useMemo } from "react";
import { StyleSheet } from "../shared-types/variants-interface";
export default function AndroidMockup(props) {
    var screenWidth = props.screenWidth, _a = props.noRoundedScreen, noRoundedScreen = _a === void 0 ? false : _a, _b = props.isLandscape, isLandscape = _b === void 0 ? false : _b, _c = props.frameColor, frameColor = _c === void 0 ? "#666666" : _c, _d = props.frameOnly, frameOnly = _d === void 0 ? false : _d, _e = props.statusbarColor, statusbarColor = _e === void 0 ? "#CCCCCC" : _e, _f = props.hideStatusBar, hideStatusBar = _f === void 0 ? false : _f, _g = props.navBar, navBar = _g === void 0 ? "swipe" : _g, _h = props.navBarColor, navBarColor = _h === void 0 ? "#CCCCCC" : _h, _j = props.transparentNavBar, transparentNavBar = _j === void 0 ? false : _j, _k = props.hideNavBar, hideNavBar = _k === void 0 ? false : _k, _l = props.transparentCamArea, transparentCamArea = _l === void 0 ? false : _l;
    return (_jsx("div", __assign({ className: props.className, style: props.containerStlye }, { children: isLandscape ? (
        // eslint-disable-next-line no-use-before-define
        _jsx(AndroidLandscape, __assign({ screenWidth: screenWidth, screenRounded: !noRoundedScreen, frameColor: frameColor, frameOnly: frameOnly, statusbarColor: statusbarColor, navigationBar: navBar, navigationBarColor: navBarColor, hideStatusBar: hideStatusBar, transparentCamArea: transparentCamArea, transparentNavigationBar: transparentNavBar, hideNavigationBar: hideNavBar }, { children: props.children }))) : (
        // eslint-disable-next-line no-use-before-define
        _jsx(AndroidPortrait, __assign({ screenWidth: screenWidth, screenRounded: !noRoundedScreen, frameColor: frameColor, frameOnly: frameOnly, statusbarColor: statusbarColor, navigationBar: navBar, navigationBarColor: navBarColor, hideStatusBar: hideStatusBar, transparentNavigationBar: transparentNavBar, hideNavigationBar: hideNavBar }, { children: props.children }))) })));
}
var getStyles = function (isLandscape, getSizeWithRatio, screenWidth, mHeight, screenRounded, frameColor, statusbarColor, navigationBarColor, frameOnly) {
    var FRAME_WIDTH = getSizeWithRatio(32);
    var HALF_FRAME_WIDTH = Math.floor(FRAME_WIDTH / 2);
    var widthAndFrame = screenWidth + FRAME_WIDTH * 2;
    var heightAndFrame = mHeight + FRAME_WIDTH * 2;
    var frameButtonSize = Math.floor(FRAME_WIDTH * 0.9);
    var frameButtonPosition = (isLandscape ? mHeight : screenWidth) + FRAME_WIDTH + HALF_FRAME_WIDTH;
    var subItemSize = getSizeWithRatio(60);
    var contPaddingRight; // for Portrait
    var contPaddingTop; // for Landscape
    if (frameOnly) {
        contPaddingRight = 0;
        contPaddingTop = 0;
    }
    else if (isLandscape) {
        contPaddingRight = 0;
        contPaddingTop = frameButtonSize - HALF_FRAME_WIDTH + 1;
    }
    else {
        contPaddingRight = frameButtonSize - HALF_FRAME_WIDTH + 1;
        contPaddingTop = 0;
    }
    var styles = StyleSheet.create({
        container: {
            display: "flex",
            position: "relative",
            boxSizing: "content-box",
            width: widthAndFrame,
            height: heightAndFrame,
            paddingTop: contPaddingTop,
            paddingRight: contPaddingRight,
        },
        frame: {
            display: "flex",
            flexDirection: "column",
            borderRadius: screenRounded ? getSizeWithRatio(140) : getSizeWithRatio(30),
            borderStyle: "solid",
            borderWidth: FRAME_WIDTH,
            borderColor: frameColor,
            overflow: "hidden",
        },
        screenCont: { display: "flex", flex: 1, overflow: "hidden" },
        screen: {
            display: "flex",
            flexDirection: isLandscape ? "row" : "column",
            position: "relative",
            width: screenWidth,
            height: mHeight,
            backgroundColor: "transparent",
            overflow: "hidden",
        },
        statusbar: {
            display: "flex",
            flexDirection: "column",
            width: isLandscape ? getSizeWithRatio(90) : "100%",
            height: isLandscape ? "100%" : getSizeWithRatio(90),
            backgroundColor: statusbarColor,
            alignItems: isLandscape ? "flex-start" : "center",
            justifyContent: isLandscape ? "center" : "flex-start",
        },
        camera: {
            width: subItemSize,
            height: subItemSize,
            borderRadius: subItemSize,
            backgroundColor: frameColor,
            marginTop: isLandscape ? 0 : getSizeWithRatio(20),
            marginLeft: isLandscape ? getSizeWithRatio(20) : 0,
        },
        // landscape only
        statusbarLand: {
            height: getSizeWithRatio(50),
            backgroundColor: statusbarColor,
        },
        navigationSwipeCont: {
            display: "flex",
            width: "100%",
            height: getSizeWithRatio(isLandscape ? 50 : 60),
            backgroundColor: navigationBarColor,
            alignItems: "center",
            justifyContent: "center",
        },
        navigationSwipeTransparentCont: {
            display: "flex",
            position: "absolute",
            bottom: 0,
            width: "100%",
            height: getSizeWithRatio(isLandscape ? 50 : 60),
            alignItems: "center",
            justifyContent: "center",
            pointerEvents: "none",
        },
        navigationSwipeBar: {
            backgroundColor: frameColor,
            borderRadius: getSizeWithRatio(100),
            width: getSizeWithRatio(350),
            height: getSizeWithRatio(15),
        },
        navigationBhrCont: {
            display: "flex",
            position: "relative",
            boxSizing: "border-box",
            flexDirection: isLandscape ? "column" : "row",
            width: isLandscape ? getSizeWithRatio(120) : "100%",
            height: isLandscape ? "100%" : getSizeWithRatio(120),
            backgroundColor: navigationBarColor,
            paddingTop: isLandscape ? getSizeWithRatio(260) : 0,
            paddingBottom: isLandscape ? getSizeWithRatio(260) : 0,
            paddingLeft: isLandscape ? 0 : getSizeWithRatio(260),
            paddingRight: isLandscape ? 0 : getSizeWithRatio(260),
            alignItems: "center",
            justifyContent: "space-between",
        },
        navigationBhrTransParent: {
            position: "absolute",
            display: "flex",
            flexDirection: isLandscape ? "column" : "row",
            boxSizing: "border-box",
            bottom: isLandscape ? undefined : 0,
            right: isLandscape ? 0 : undefined,
            width: isLandscape ? getSizeWithRatio(120) : "100%",
            height: isLandscape ? "100%" : getSizeWithRatio(120),
            paddingTop: isLandscape ? getSizeWithRatio(260) : 0,
            paddingBottom: isLandscape ? getSizeWithRatio(260) : 0,
            paddingLeft: isLandscape ? 0 : getSizeWithRatio(260),
            paddingRight: isLandscape ? 0 : getSizeWithRatio(260),
            alignItems: "center",
            justifyContent: "space-between",
            pointerEvents: "none",
        },
        volumePortrait: {
            position: "absolute",
            borderRadius: FRAME_WIDTH,
            top: getSizeWithRatio(420),
            left: frameButtonPosition,
            width: frameButtonSize,
            height: getSizeWithRatio(330),
            backgroundColor: frameColor,
        },
        volumeLandscape: {
            position: "absolute",
            borderRadius: FRAME_WIDTH,
            left: getSizeWithRatio(420),
            bottom: frameButtonPosition,
            width: getSizeWithRatio(330),
            height: frameButtonSize,
            backgroundColor: frameColor,
        },
        powerPortrait: {
            position: "absolute",
            borderRadius: FRAME_WIDTH,
            top: getSizeWithRatio(900),
            left: frameButtonPosition,
            width: frameButtonSize,
            height: getSizeWithRatio(180),
            backgroundColor: frameColor,
        },
        powerLandscape: {
            position: "absolute",
            borderRadius: FRAME_WIDTH,
            left: getSizeWithRatio(900),
            bottom: frameButtonPosition,
            width: getSizeWithRatio(180),
            height: frameButtonSize,
            backgroundColor: frameColor,
        },
        triangle: {
            width: 0,
            height: 0,
            borderTopWidth: subItemSize / 2,
            borderTopStyle: "solid",
            borderTopColor: "transparent",
            borderBottomWidth: subItemSize / 2,
            borderBottomStyle: "solid",
            borderBottomColor: "transparent",
            borderRightWidth: subItemSize,
            borderRightStyle: "solid",
            borderRightColor: frameColor,
            opacity: 0.6,
        },
        circle: {
            width: subItemSize,
            height: subItemSize,
            borderRadius: subItemSize,
            backgroundColor: frameColor,
            opacity: 0.6,
        },
        square: {
            width: subItemSize,
            height: subItemSize,
            borderRadius: subItemSize / 10,
            backgroundColor: frameColor,
            opacity: 0.6,
        },
        cameraFullScreen: {
            position: "absolute",
            alignSelf: "center",
            verticalAlign: isLandscape ? "middle" : undefined,
            top: isLandscape ? undefined : getSizeWithRatio(20),
            left: isLandscape ? getSizeWithRatio(20) : undefined,
            width: subItemSize,
            height: subItemSize,
            borderRadius: subItemSize,
            backgroundColor: frameColor,
        },
    });
    return styles;
};
function AndroidPortrait(props) {
    var screenWidth = props.screenWidth, screenRounded = props.screenRounded, frameColor = props.frameColor, frameOnly = props.frameOnly, statusbarColor = props.statusbarColor, navigationBar = props.navigationBar, navigationBarColor = props.navigationBarColor, hideStatusBar = props.hideStatusBar, hideNavigationBar = props.hideNavigationBar, transparentNavigationBar = props.transparentNavigationBar;
    var styles = useMemo(function () {
        var getSizeWithRatio = function (size) {
            var sizeRatio = Math.floor((screenWidth * size) / 1080);
            return Math.max(sizeRatio, 1);
        };
        var mHeight = Math.floor((screenWidth / 9) * 19.5);
        return getStyles(false, getSizeWithRatio, screenWidth, mHeight, screenRounded, frameColor, statusbarColor, navigationBarColor, frameOnly);
    }, [screenWidth, screenRounded, frameColor, statusbarColor, navigationBarColor, frameOnly]);
    return (_jsxs("div", __assign({ style: styles.container }, { children: [_jsx("div", __assign({ style: styles.frame }, { children: _jsxs("div", __assign({ style: styles.screen }, { children: [hideStatusBar === false && (_jsx("div", __assign({ style: styles.statusbar }, { children: _jsx("div", { style: styles.camera }) }))), _jsx("div", __assign({ style: styles.screenCont }, { children: props.children })), hideStatusBar && _jsx("div", { style: styles.cameraFullScreen }), hideNavigationBar === false &&
                            navigationBar === "swipe" &&
                            (transparentNavigationBar ? (_jsx("div", __assign({ style: styles.navigationSwipeTransparentCont }, { children: _jsx("div", { style: styles.navigationSwipeBar }) }))) : (_jsx("div", __assign({ style: styles.navigationSwipeCont }, { children: _jsx("div", { style: styles.navigationSwipeBar }) })))), hideNavigationBar === false &&
                            navigationBar === "bhr" &&
                            (transparentNavigationBar ? (_jsxs("div", __assign({ style: styles.navigationBhrTransParent }, { children: [_jsx("div", { style: styles.triangle }), _jsx("div", { style: styles.circle }), _jsx("div", { style: styles.square })] }))) : (_jsxs("div", __assign({ style: styles.navigationBhrCont }, { children: [_jsx("div", { style: styles.triangle }), _jsx("div", { style: styles.circle }), _jsx("div", { style: styles.square })] })))), hideNavigationBar === false &&
                            navigationBar === "rhb" &&
                            (transparentNavigationBar ? (_jsxs("div", __assign({ style: styles.navigationBhrTransParent }, { children: [_jsx("div", { style: styles.square }), _jsx("div", { style: styles.circle }), _jsx("div", { style: styles.triangle })] }))) : (_jsxs("div", __assign({ style: styles.navigationBhrCont }, { children: [_jsx("div", { style: styles.square }), _jsx("div", { style: styles.circle }), _jsx("div", { style: styles.triangle })] }))))] })) })), !frameOnly && (_jsxs(_Fragment, { children: [_jsx("div", { style: styles.volumePortrait }), _jsx("div", { style: styles.powerPortrait })] }))] })));
}
function AndroidLandscape(props) {
    var screenWidth = props.screenWidth, screenRounded = props.screenRounded, frameColor = props.frameColor, frameOnly = props.frameOnly, statusbarColor = props.statusbarColor, navigationBar = props.navigationBar, navigationBarColor = props.navigationBarColor, hideStatusBar = props.hideStatusBar, hideNavigationBar = props.hideNavigationBar, transparentNavigationBar = props.transparentNavigationBar, transparentCamArea = props.transparentCamArea;
    var styles = useMemo(function () {
        var getSizeWithRatio = function (size) {
            var sizeRatio = Math.floor((screenWidth * size) / 2340);
            return Math.max(sizeRatio, 1);
        };
        var mHeight = Math.floor((screenWidth / 19.5) * 9);
        return getStyles(true, getSizeWithRatio, screenWidth, mHeight, screenRounded, frameColor, statusbarColor, navigationBarColor, frameOnly);
    }, [screenWidth, screenRounded, frameColor, statusbarColor, navigationBarColor, frameOnly]);
    return (_jsxs("div", __assign({ style: styles.container }, { children: [_jsx("div", __assign({ style: styles.frame }, { children: _jsxs("div", __assign({ style: styles.screen }, { children: [transparentCamArea === false && (_jsx("div", __assign({ style: styles.statusbar }, { children: _jsx("div", { style: styles.camera }) }))), _jsxs("div", __assign({ style: { display: "flex", flexDirection: "column", flex: 1 } }, { children: [hideStatusBar === false && _jsx("div", { style: styles.statusbarLand }), _jsx("div", __assign({ style: styles.screenCont }, { children: props.children })), hideNavigationBar === false &&
                                    navigationBar === "swipe" &&
                                    transparentNavigationBar === false && (_jsx("div", __assign({ style: styles.navigationSwipeCont }, { children: _jsx("div", { style: styles.navigationSwipeBar }) })))] })), transparentCamArea && _jsx("div", { style: styles.cameraFullScreen }), hideNavigationBar === false &&
                            navigationBar === "swipe" &&
                            transparentNavigationBar && (_jsx("div", __assign({ style: styles.navigationSwipeTransparentCont }, { children: _jsx("div", { style: styles.navigationSwipeBar }) }))), hideNavigationBar === false &&
                            navigationBar === "bhr" &&
                            (transparentNavigationBar ? (_jsxs("div", __assign({ style: styles.navigationBhrTransParent }, { children: [_jsx("div", { style: styles.square }), _jsx("div", { style: styles.circle }), _jsx("div", { style: styles.triangle })] }))) : (_jsxs("div", __assign({ style: styles.navigationBhrCont }, { children: [_jsx("div", { style: styles.square }), _jsx("div", { style: styles.circle }), _jsx("div", { style: styles.triangle })] })))), hideNavigationBar === false &&
                            navigationBar === "rhb" &&
                            (transparentNavigationBar ? (_jsxs("div", __assign({ style: styles.navigationBhrTransParent }, { children: [_jsx("div", { style: styles.triangle }), _jsx("div", { style: styles.circle }), _jsx("div", { style: styles.square })] }))) : (_jsxs("div", __assign({ style: styles.navigationBhrCont }, { children: [_jsx("div", { style: styles.triangle }), _jsx("div", { style: styles.circle }), _jsx("div", { style: styles.square })] }))))] })) })), !frameOnly && (_jsxs(_Fragment, { children: [_jsx("div", { style: styles.volumeLandscape }), _jsx("div", { style: styles.powerLandscape })] }))] })));
}
